### ISSUE TITLE
mkWindowsApp lock to prevent multiple `winAppInstall` launches

### ISSUE THREAD

**Issue by crabdancing** [emmanuelrosa/erosanix#22]
If you set up a GUI icon, a lot of users will have the instinct to try to open it again if it seems slow. This results in potentially multiple installer phase instances being launched at the same time. Having some sort of file lock set during `winAppInstall` would prevent this. While it's certainly possible to set this up outside of `mkWindowsApp`, the fact that it's useful across all `mkWindowsApp` packages means that it would make sense to add this feature to `mkWindowsApp` itself.

**Comment by emmanuelrosa** [comment]
Firstly, thank you for reporting your insights.

I developed the initial version of `mkWindowsApp` about 2 years ago, and it's only recently that I've noticed others have started to use it. Naturally, this means bugs will be discovered now :)

I'll take a look at this, and the other issue you reported.

Currently, there's a sort-of locking mechanism but it only takes effect after a complete installation. All it does is skip certain steps, so it's not a lock in the sense you've described.
